tests/playbooks/tests_wireless_and_network_restart.yml

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Play for testing wifi and network restart
+  hosts: all
+  vars:
+    interface: wlan0
+    profile: "{{ interface }}"
+    package: NetworkManager-wifi
+    wifi_restart_network: false
+    network_connections_result_error:
+      NetworkManager needs to be restarted to be able to proceed
+      because wireless and team interfaces are defined. This might
+      disturb the connectivity of the managed system. Please set
+      `network_allow_restart` to `true` if you are prepared for this.
+      Notice that the necessary action is to install NetworkManager-wifi or
+      NetworkManager-team plugin and to restart NetworkManager.
+    lsr_fail_debug:
+      - __network_connections_result
+  tasks:
+    - name: Show playbook name
+      debug:
+        msg: "this is: playbooks/tests_wireless_and_network_restart.yml"
+      tags:
+        - always
+
+    - name: Check if rpm ostree system - cannot test
+      meta: end_host
+      when: __network_is_ostree | d(false)
```

Comment:
What's the reason this cannot be tested on an rpm ostree system?

Reply:
https://github.com/linux-system-roles/network/blob/main/README-ostree.md#rpm-ostree, " the /usr filesystem is read-only, and the role cannot install packages".

Reply:
The problem is that we don't have a good way currently to pass in the custom yum repos/copr repos that are used by these tests.  If you could build an image with all of the repos and packages, then I think the tests should work.

Reply:
> [`main`/README-ostree.md#rpm-ostree](https://github.com/linux-system-roles/network/blob/main/README-ostree.md?rgh-link-date=2024-03-13T13%3A32%3A38Z#rpm-ostree), " the /usr filesystem is read-only, and the role cannot install packages".

This is not a sufficient explanation. Because this still leaves the question what's the reason that this cannot be tested? The role needs to have a defined behavior when running against ostree systems.



> The problem is that we don't have a good way currently to pass in the custom yum repos/copr repos that are used by these tests. If you could build an image with all of the repos and packages, then I think the tests should work.

Maybe I am missing something. If an ostree system is missing the wifi or team subpackage (or even some NM or Nmstate package), then the role needs to handle this with a proper error message (if we intend to support ostree targets). If the role handles this correctly.

Reply:
The problem is that the test wants to remove the package `NetworkManager-wifi` then reinstall it later - packages cannot be removed from an ostree system - the only thing package management can do at runtime is to verify that the given package exists (or is absent).

If the test needs to verify the package `NetworkManager-wifi` is absent, then call the role to install it, then verify that the package `NetworkManager-wifi` is present, then at least this part of the test is not applicable for ostree systems.  If we really want this test to be run on ostree systems, then we need to modify the test such that the testing of the package absence and presence is skipped on ostree systems.  I don't know if that will substantially change what this test is doing.

Reply:
From what I understand so far, for the ostree case, if the package is installed, the test should be skipped. If the package is missing, the test should assert that the role produces a descriptive error message.

Reply:
> From what I understand so far, for the ostree case, if the package is installed, the test should be skipped.

Then the test should always be skipped in our testing frameworks, because we always build the image with all of the packages needed for every runtime and testing configuration: https://github.com/linux-system-roles/network/blob/main/.ostree/packages-runtime.txt

> If the package is missing, the test should assert that the role produces a descriptive error message.

The package should never be missing in our ostree testing.

Reply:
Which testing frameworks remain to test package installation for the role? Which systems would still run the test?

Reply:
> Which testing frameworks remain to test package installation for the role?

Our CI system, local developer testing via `tox -e qemu...`, and all of our downstream testing.

> Which systems would still run the test?

All of the above.

That is - we are still running all tests, in a few different ways, with "regular" systems that can install and remove packages at runtime.

Reply:
I do not follow, previously you wrote "Then the test should always be skipped in our testing frameworks," - when exactly will it be skipped and when not? Just considering automated CI.

Reply:
> I do not follow, previously you wrote "Then the test should always be skipped in our testing frameworks," - when exactly will it be skipped and when not? Just considering automated CI.

We currently do not have automated CI for ostree.

Because of the way we build images for ostree testing, the `NetworkManager-wifi` package will always be installed in the image.

If a customer builds an ostree image, then runs the network role against it, and they specify some parameter that requires an additional package that is not built into the ostree image, then the `package:` task will print a message like "The package XXX is not found".

Reply:
ok, good enough.

Reply:
What's missing is to explain this in the code.

Reply:
> What's missing is to explain this in the code.

How?  As a code comment in tasks/main.yml for the various `package:` tasks?

Reply:
I was thinking of changing `name: Check if rpm ostree system - cannot test` to something like `name: Cannot test ostree system because ... `

Reply:
If there are multiple places, maybe it is also worth to write this in one file and include it in all places. Doing this in a follow-up PR is fine.

Reply:
tracked by https://github.com/linux-system-roles/network/issues/681

Reply:
I meant doing this at other places and/or using a task file for all those places is something to do in a different PR. The task in this PR needs a good name, still.

Reply:
I renamed the task into following.

```
    - name: Cannot test ostree systems because the package can not be removed
        or installed - 'NetworkManager-wifi'
```